renku/core/workflow/converters/cwl.py

```diff
@@ -94,38 +94,64 @@ def workflow_format(self):
 
     @hookimpl
     def workflow_convert(
-        self, workflow: Union[CompositePlan, Plan], basedir: Path, output: Optional[Path], output_format: Optional[str]
-    ):
+        self,
+        workflow: Union[CompositePlan, Plan],
+        basedir: Path,
+        output: Optional[Path],
+        output_format: Optional[str],  # only YAML is generated
+        resolve_paths: bool,  # if True will make paths absolute and resolve symlinks
+        nest_workflows: Optional[bool],  # if True, generate a single workflow spec and file
+    ) -> str:
         """Converts the specified workflow to CWL format."""
-        filename = None
+        if nest_workflows is None:
+            nest_workflows = False
+
         if output:
             if output.is_dir():
                 tmpdir = output
+                filename = None
             else:
                 tmpdir = output.parent
                 filename = output
         else:
             tmpdir = Path(tempfile.mkdtemp())
 
         if isinstance(workflow, CompositePlan):
-            path = CWLExporter._convert_composite(
-                workflow, tmpdir, basedir, filename=filename, output_format=output_format
-            )
+            cwl_workflow = CWLExporter._convert_composite(
+                workflow, basedir, resolve_paths=resolve_paths
+            )  # type: ignore
```

Comment:
don't use `# type: ignore` unless absolutely necessary.

Doing `cwl_workflow: Union[cwl.Workflow, CommandLineTool]` above before the if should resolve the type error.